src/cmap/connection.ts

```diff
@@ -430,7 +436,9 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
   ): AsyncGenerator<MongoDBResponse> {
     this.throwIfAborted();
 
-    if (typeof options.socketTimeoutMS === 'number') {
+    if (options.timeoutContext?.csotEnabled()) {
+      this.socket.setTimeout(0);
+    } else if (typeof options.socketTimeoutMS === 'number') {
       this.socket.setTimeout(options.socketTimeoutMS);
     } else if (this.socketTimeoutMS !== 0) {
       this.socket.setTimeout(this.socketTimeoutMS);
```

Comment:
Currently, this will timeout the `'data'` event we wait on which represents a whole message whereas the `socket.setTimeout` represents chunks, for CSOT we do need to change the way we timeout a message from how it was done previously.

We could consider changing the old way to work similar to CSOT.

Reply:
Oh I see, yea that would work, how do you intend we resolve the `options.socketTimeoutMS` vs the one on the class `this.socketTimeoutMS` currently there's this fallback behavior, I need to hand the context an instance of the connection class and the options.

Are you thinking something along these lines:
```ts
const t = context.socketTimeoutMS(this, options) // CSOT returns 0, legacy resolves options vs connection
socket.setTimeout(t)
```